test/src/unit-udt.cpp

```diff
@@ -311,30 +311,6 @@ TEST_CASE("basic usage" * doctest::test_suite("udt"))
             person_json["name"].get_to(name).m_val = "new name";
             CHECK(name.m_val == "new name");
         }
-
```

Comment:
I do not think this test should be removed. Can't we detect the flag?

Reply:
It's been a while since I've worked on this PR, I'll try to make both cohabit. That's also why I've put `JSON_ExplicitConversions=ON` everywhere.

Reply:
One annoying thing is that code using implicit conversions will not build when explicit conversions are enabled, I'd like to find a way to avoid sprinkling `ifdef` everywhere in the tests.